druid/src/contexts.rs

```diff
@@ -142,6 +152,169 @@ pub struct PaintCtx<'a, 'b, 'c> {
     pub(crate) depth: u32,
 }
 
+/// The state of a widget and its global context.
+pub struct State<'a, 'b> {
+    #[allow(dead_code)]
```

Comment:
Currently the only method using `State` is `set_origin`, which does not the context_state, but it felt wrong to only add the bits of state i needed to this struct.

Reply:
That feels wrong to me. YAGNI comes to mind. We are paying a real cost both in terms of compile time and in terms of mental space when reading the code with the extra lifetime names. Yet we aren't getting any value *right now* out of this.

I understand the idea of this being maybe useful in the future, yet this type of stuff usually ends up being a premature thing to do.

I would rather not include this as live code. We could leave it in as a commented-out line with a comment mentioning that this could be enabled when there is an actual need. However the lifetimes of the `State` struct would still need to be redone.

Reply:
Sounds good :)